File: 1.两数之和.java
/*
 * @lc app=leetcode.cn id=1 lang=java
 *
 * [1] 两数之和
 */

// @lc code=start
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> map = new HashMap<>();
        int n = nums.length;
        for(int i=0;i<n;i++){
            if(map.containsKey(target-nums[i]) &&map.get(target-nums[i])!=i){
                return new int[]{i,map.get(target-nums[i])};
            }else{
                map.put(nums[i],i);
            }
        }
        
        return null;
    }
}
// @lc code=end

File: 1.两数之和.py
#
# @lc app=leetcode.cn id=1 lang=python3
# [1] 两数之和
#

# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map={} #val index
        for i,n in enumerate(nums):
            diff=target-n
            if diff in map:
                return [map[diff],i]
            map[n]=i
        return 
# @lc code=end

